### PR TITLE
VXLAN changes

### PR DESCRIPTION
**Pull request by leodotcloud**
@ibuildthecloud @alena1108 : Initial changes, still more changes and corner cases to be handled. Creating the PR to share the high level architecture.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -113,27 +133,61 @@ func appMain(ctx *cli.Context) error {
...
+	db.Reload()
+
+	var overlay backend.Backend
+	if backendToUse == backendNameVxlan {
+		overlay, _ = vxlan.NewOverlay("", db)
+		overlay.Start(true, "")
+	} else {
+		overlay = ipsec.NewOverlay(ctx.GlobalString("ipsec-config"), db)
+		if !ctx.GlobalBool("gcm") {
+			// TODO: @ibuildthecloud: Fix this line later, commenting for now
```
Comment (ibuildthecloud) [inline main.go]: What's going on here?

File: mdchandler/metadata_change_handler.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	mc      metadata.Client
+}
+
+// NewMetadataChangeHandler is used to create a OnChange
+// handler for Meatadta
+func NewMetadataChangeHandler(b backend.Backend) *MetadataChangeHandler {
+	mc, err := metadata.NewClientAndWait(metadataURL)
+	if err != nil {
+		logrus.Errorf("couldn't create metadata client: %v", err)
+		return nil
```
Comment (ibuildthecloud) [inline mdchandler/metadata_change_handler.go:27]: This will just cause a panic

File: main.go (hunk 1 above)
Comment (leodotcloud) [reply main.go]: Fixed this.

File: mdchandler/metadata_change_handler.go (hunk 2 above)
Comment (leodotcloud) [reply mdchandler/metadata_change_handler.go:27]: Will fix it in the later changes.

**Pull request merged**